### PR TITLE
[backport] ROCm: Support hipCUB/rocPRIM

### PR DESCRIPTION
**Pull request by emcastillo**
Backport of #4027

### CODE REVIEW

**Comment by emcastillo** [comment]
[automatic post] Jenkins, test this please.

File: cupy/_environment.py [hunk 1]
```diff
@@ -10,10 +10,14 @@
 import sys
 import warnings
 
+from cupy_backends.cuda.api import runtime
+
```
Comment (emcastillo) [inline cupy/_environment.py:14]:
```suggestion
```

**Comment by emcastillo** [comment]
Jenkins, test this please

File: cupy/_statistics/histogram.py [hunk 2]
```diff
@@ -8,6 +8,8 @@
 from cupy.core import _accelerator
 from cupy.cuda import cub
 from cupy.cuda import common
+from cupy.cuda import runtime
+
```
Comment (toslunar) [inline cupy/_statistics/histogram.py:12]:
```
./cupy/_statistics/histogram.py:15:1: E303 too many blank lines (3)
Error: Process completed with exit code 1.
```

Comment (emcastillo) [inline cupy/_statistics/histogram.py:12]:
```suggestion¥
```

Comment (emcastillo) [inline cupy/_statistics/histogram.py:12]:
```suggestion
```

**Comment by emcastillo** [comment]
Jenkins, test this please

**Pull request merged**